### PR TITLE
Da/changed ethers to web3

### PR DESCRIPTION
**Pull request by axelrod-blox**

### CODE REVIEW

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 1]
```diff
@@ -285,44 +287,47 @@ class OperatorStore extends BaseStore {
...
-                await this.refreshOperatorsAndClusters(resolve, true);
-                break;
-              } else {
-                console.log('Operator is still not updated in API..');
+        await contractInstance.methods.setOperatorWhitelist(operatorId, address).send({ from: walletStore.accountAddress, gas: gasLimit })
+          .on('receipt', async (receipt: any) => {
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              let iterations = 0;
+              while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
```
Comment (IlyaVi) [inline src/app/common/stores/applications/SsvWeb/Operator.store.ts:301]: this while loop and all while loops structured the same are meaningless
code after if statement is unreachable - thus redundant. Please find every appearance of this code structure and simplify it to a single call to `await this.refreshOperatorsAndClusters(resolve, true);`
for the reference see this link [break in while loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/break#examples)

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 2]
```diff
@@ -346,67 +351,70 @@ class OperatorStore extends BaseStore {
...
-              if (iterations >= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
+        contract.methods.cancelDeclaredOperatorFee(operatorId).send({ from: walletStore.accountAddress, gas: gasLimit })
+          .on('receipt', async (receipt: any) => {
+            // eslint-disable-next-line no-prototype-builtins
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              ApiParams.initStorage(true);
+              console.debug('Contract Receipt', receipt);
+              let iterations = 0;
+              while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
```
Comment (IlyaVi) [inline src/app/common/stores/applications/SsvWeb/Operator.store.ts:375]: same issue as before

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 3]
```diff
@@ -507,61 +515,68 @@ class OperatorStore extends BaseStore {
...
-                break;
+        await contractInstance.methods.declareOperatorFee(operatorId, formattedFee).send({
+          from: walletStore.accountAddress,
+          gas: gasLimit,
+        })
+          .on('receipt', async (receipt: any) => {
+            // eslint-disable-next-line no-prototype-builtins
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              let iterations = 0;
```
Comment (IlyaVi) [inline src/app/common/stores/applications/SsvWeb/Operator.store.ts:526]: same issue

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 4]
```diff
@@ -570,47 +585,53 @@ class OperatorStore extends BaseStore {
...
+        await contractInstance.methods.reduceOperatorFee(operatorId, formattedFee).send({
+          from: walletStore.accountAddress,
+          gas: gasLimit,
+        })
+          .on('receipt', async (receipt: any) => {
+            // eslint-disable-next-line no-prototype-builtins
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              let iterations = 0;
+              while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
```
Comment (IlyaVi) [inline src/app/common/stores/applications/SsvWeb/Operator.store.ts:597]: same issue

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 5]
```diff
@@ -621,68 +642,74 @@ class OperatorStore extends BaseStore {
...
+        await walletStore.setterContract.methods.executeOperatorFee(operatorId).send({
+          from: walletStore.accountAddress,
+          gas: gasLimit,
+        })
+          .on('receipt', async (receipt: any) => {
+            // eslint-disable-next-line no-prototype-builtins
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              let iterations = 0;
+              while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
```
Comment (IlyaVi) [inline src/app/common/stores/applications/SsvWeb/Operator.store.ts:668]: same issue

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 6]
```diff
@@ -696,50 +723,60 @@ class OperatorStore extends BaseStore {
...
+          from: walletStore.accountAddress,
+          gas: gasLimit,
+        })
+          .on('receipt', async (receipt: any) => {
+            // eslint-disable-next-line no-prototype-builtins
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              ApiParams.initStorage(true);
+              console.debug('Contract Receipt', receipt);
+              let iterations = 0;
```
Comment (IlyaVi) [inline src/app/common/stores/applications/SsvWeb/Operator.store.ts:743]: same issue

File: src/app/common/stores/applications/SsvWeb/SSV.store.ts [hunk 7]
```diff
@@ -388,21 +427,53 @@ class SsvStore extends BaseStore {
...
   }
-
   /**
    * Get new account burn rate
    */
   getNewAccountBurnRate(oldOperatorsFee: number, newOperatorsFee: number): number {
     return this.accountBurnRate - oldOperatorsFee + newOperatorsFee;
   }
 
+  // /**
```
Comment (IlyaVi) [inline src/app/common/stores/applications/SsvWeb/SSV.store.ts:444]: remove

**Review by IlyaVi — APPROVED** [review]

**Review by meshin-dev — APPROVED** [review]

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts (hunk 3 above)
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts:514]: +

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts (hunk 4 above)
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts:581]: +

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts (hunk 5 above)
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts:648]: +

File: src/app/common/stores/applications/SsvWeb/SSV.store.ts (hunk 7 above)
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/SSV.store.ts]: +

File: src/app/common/stores/applications/SsvWeb/Operator.store.ts [hunk 8]
```diff
@@ -285,44 +287,46 @@ class OperatorStore extends BaseStore {
...
+        })
+          .on('receipt', async (receipt: any) => {
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              let iterations = 0;
+              while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
+                const operator = await Operator.getInstance().getOperator(operatorId);
+                const changed = operator.address_whitelist.toString() === address.toString();
+                if (changed) {
+                  break;
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/stores/applications/SsvWeb/Operator.store.ts:308]: done

**Pull request merged**